src/util/mpir_pmi.c

```diff
@@ -257,6 +265,11 @@ const char *MPIR_pmi_job_id(void)
     return (const char *) pmi_kvs_name;
 }
 
+const char *MPIR_pmi_hostname(void)
```

Comment:
My only concern is in PMIx we have the `PMIX_HOSTNAME` attribute, which basically wraps `gethostname` as a convenience. It would be good to avoid confusion, probably by changing the naming. MPICH has remnants of `MPICH_INTERFACE_HOSTNAME` in the code. Maybe `PMI_I[NTER]FACE_HOSTNAME`?

Reply:
I think `PMIX_HOSTNAME` is of the same semantics as the proposed `PMI_HOSTNAME` here. The intention is for the process launcher to inform the process the local hostname used to launch. Whether it is from `gethostname` or via command line option is really an implementation issue.

Reply:
OK, fair enough then.